src/main/kotlin/io/emeraldpay/dshackle/config/ChainsConfig.kt

```diff
@@ -0,0 +1,63 @@
+package io.emeraldpay.dshackle.config
+
+import io.emeraldpay.dshackle.Chain
+import java.lang.IllegalStateException
+
+data class ChainsConfig(private val chains: Map<Chain, RawChainConfig>, val currentDefault: RawChainConfig?) {
+    companion object {
+        @JvmStatic
+        fun default(): ChainsConfig = ChainsConfig(emptyMap(), RawChainConfig.default())
+    }
+
+    data class RawChainConfig(val syncingLagSize: Int?, val laggingLagSize: Int?) {
+        companion object {
+            @JvmStatic
+            fun default() = RawChainConfig(6, 1)
+        }
+    }
+
+    data class ChainConfig(val syncingLagSize: Int, val laggingLagSize: Int) {
+        companion object {
+            @JvmStatic
+            fun default() = ChainConfig(6, 1)
+        }
+    }
+
+    fun resolve(chain: Chain): ChainConfig {
+        val default = currentDefault ?: panic()
```

Comment:
Maybe make `currentDefault` not null to avoid this check?

Reply:
We need to have a null types version for parsing and easy merging later